federated-learning-controller/README.md

```diff
@@ -1,76 +1,168 @@
 
----
+This Kubernetes controller automates the deployment and management of federated learning in an Open Cluster Management environment. The `FederatedLearning` Custom Resource Definition (CRD) provides a unified open interface for integrating frameworks such as Flower, OpenFL, and NVIDIA FLARE. It leverages Kubernetes-native resources to provision servers, launch clients, and orchestrate the training lifecycle across a multicluster environment.
 
-## Bring Federated Learning Across Multiple Clusters
+![Controller Architecture](./assets/images/architecture.png)
 
-This controller enables federated learning across a **multicluster environment** without the need for manual orchestration. The only requirement is to **containerize your workload**, making it compatible with different federated learning frameworks.  
+--- 
 
-The controller reconciles `FederatedLearning` instances, managing the lifecycle of **server** and **client** components that handle model training and aggregation. To integrate with the provided interface, servers and clients should follow the expected startup patterns.  
+## Getting Started
 
-For example, see: [Flower PyTorch App](./examples/flower/Makefile)
+### Prerequisites
 
-- **Server**
+Ensure the following tools are installed:
 
-  Creates a Kubernetes Job using a built-in manifest template, expected to start with:
+- [`kubectl`](https://kubernetes.io/docs/reference/kubectl/)
+- [`kustomize`](https://kubectl.docs.kubernetes.io/installation/kustomize/)
+- [`kind`](https://kind.sigs.k8s.io/) (version > v0.9.0 recommended)
+- [`make`](https://www.gnu.org/software/make/) for build automation
 
-  ```bash
-  server --num-rounds <number-of-rounds> ...
-  ```
+Optional (for container image building):
 
-- **Client**  
+- Podman or Docker
+- Go (version 1.19 or later)
 
-  Creates a Kubernetes Job via **ManifestWorks** from the hub cluster, expected to start with: 
+--- 
 
-  ```bash
-  client --data-config <data-configuration> --server-address <aggregator-address> ...
-  ```
+### Set Up the Environment
 
---- 
+#### 1. Install `clusteradm`
 
-## Getting Started
+```bash
+$ curl -L https://raw.githubusercontent.com/open-cluster-management-io/clusteradm/main/install.sh | bash
+```
+
+#### 2. Create hub and managed clusters with `kind`
 
-### Running Locally
+```bash
+$ curl -L https://raw.githubusercontent.com/open-cluster-management-io/OCM/main/solutions/setup-dev-environment/local-up.sh | bash
+```
 
-**Install the CRDs into the cluster:**
+#### 3. Verify cluster setup
 
-```sh
-make install
+```bash
+$ kubectl get mcl
 ```
 
-**Run the controller locally:**
+Example output:
 
-```sh
-make run
+```bash
+NAME       HUB ACCEPTED   MANAGED CLUSTER URLS                  JOINED   AVAILABLE   AGE
+cluster1   true           https://cluster1-control-plane:6443   True     True        2m
+cluster2   true           https://cluster2-control-plane:6443   True     True        3m
```

Comment:
Nit. In the later section of the doc, you use the following style and I feel like it's a bit cleaner. Maybe do the same here (and the other places that apply). Up to you.

 Example - client in managed cluster

```
  $ kubectl get pods -n open-cluster-management
  NAME                                     READY   STATUS      RESTARTS   AGE
  federated-learning-sample-client-75sc8   0/1     Completed   0          10m
```

Reply:
I modified them align with the "cleaner" one